craft_application/commands/lifecycle.py

```diff
@@ -367,13 +367,31 @@ def _run(
         """Run the pack command."""
         if step_name not in ("pack", None):
             raise RuntimeError(f"Step name {step_name} passed to pack command.")
+
+        shell = getattr(parsed_args, "shell", False)
+        shell_after = getattr(parsed_args, "shell_after", False)
+        debug = getattr(parsed_args, "debug", False)
+
+        parsed_args.shell = False
```

Comment:
This is a bit counterintuitive, so maybe a comment like this would be helpful:

```suggestion
        # Prevent the steps in the prime command from using `--shell` or `--shell-after`
        parsed_args.shell = False
```

Reply:
definitely, I meant to add a comment but forgot. Thanks!